external/vcm/vcm/derived_mapping.py

```diff
@@ -360,3 +360,43 @@ def water_vapor_path(self):
     return da.assign_attrs(
         {"long_name": "column integrated water vapor", "units": "mm"}
     )
+
+
+@DerivedMapping.register(
+    "total_sky_net_radiative_flux_at_top_of_atmosphere",
+    required_inputs=[
+        "total_sky_downward_shortwave_flux_at_top_of_atmosphere",
+        "total_sky_upward_shortwave_flux_at_top_of_atmosphere",
+        "total_sky_upward_longwave_flux_at_top_of_atmosphere",
+    ],
+)
+def total_sky_net_radiative_flux_at_top_of_atmosphere(self):
+    result = (
+        self["total_sky_downward_shortwave_flux_at_top_of_atmosphere"]
+        - self["total_sky_upward_shortwave_flux_at_top_of_atmosphere"]
+        - self["total_sky_upward_longwave_flux_at_top_of_atmosphere"]
+    )
+    if "storage_of_internal_energy_path_due_to_fine_res_temperature_nudging" in self:
```

Comment:
This seems like a modeling assumption specific to your ML model. Is this appropriate for the general use implied by including it in `vcm`?

Reply:
Can't actually register in fv3fit. Alternatively I could define this quantity in the fine-res loader.

Reply:
Are you sure this won't work? We just need to sure the "registration" is performed when needed (prog run, offline, training). Importing `fv3fit` should be enough to register these. Alternatively you could add it to the mapper (or define them as a tensor_transform).

Reply:
I did try it, and at least when using the `DerivedMapping` (imported from vcm) within a pytest in fv3fit, it did not have the quantities registered within fv3fit available. Maybe I needed to do some additional fv3fit import. However, I'm not sure that import would be guaranteed to happen if just using loaders anyway. I moved the TOA flux calculation to the fine-res mapper.

Reply:
Maybe it's a moot point, but I think the pattern should work...I've used it before. If you are interested, I could help you debug if you point me to the test code. The module containing the code was probably not being imported.

Reply:
I deleted the test code. I think I'm just going to merge as is and move on.

Reply:
No problem.